scene/main/viewport.cpp

```diff
@@ -292,6 +292,9 @@ void Viewport::_sub_window_grab_focus(Window *p_window) {
 		Window *this_window = Object::cast_to<Window>(this);
 		if (this_window) {
 			this_window->_event_callback(DisplayServer::WINDOW_EVENT_FOCUS_IN);
+		} else if (get_window_id() == DisplayServer::INVALID_WINDOW_ID) {
+			this_window = get_window();
+			this_window->_event_callback(DisplayServer::WINDOW_EVENT_FOCUS_IN);
 		}
```

Comment:
Nitpick:
```suggestion
		Window *this_window = Object::cast_to<Window>(this);
		if (!this_window && get_window_id() == DisplayServer::INVALID_WINDOW_ID) {
			this_window = get_window();
		}
		if (this_window) {
			this_window->_event_callback(DisplayServer::WINDOW_EVENT_FOCUS_IN);
		}
```
Avoids repeating what to do one you have an actual window (which could be more than a single call in the future), and prevents a crash if by any chance `get_window()` returns `nullptr`.

Reply:
Thanks for the feedback. I also applied the same principle to the focus out callback some lines under